mk/Common.mk

```diff
@@ -231,123 +261,157 @@ include $(CFG_FLAGS_MAKE)
-PULP_YES         = $(call IS_YES,$(PULP))
-COREV_YES        = $(call IS_YES,$(COREV))
-LLVM_YES         = $(call IS_YES,$(LLVM))
-
-ifeq ($(shell $(CORE_V_VERIF)/mk/toolchain_check.sh $(GNU_YES) $(PULP_YES) $(COREV_YES) $(LLVM_YES)),1)
-$(error Multiple toolchains are enabled: GNU=${GNU_YES} PULP=${PULP_YES} COREV=${COREV_YES} LLVM=${LLVM_YES})
-endif
-
-RISCV             = $(CV_SW_TOOLCHAIN)
-RISCV_PREFIX      = riscv32-$(CV_SW_VENDOR)-elf-
-RISCV_EXE_PREFIX  = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC          = gcc
-RISCV_MARCH       = $(call RESOLVE_FLAG2,$(TEST_RISCV_MARCH),$(CV_SW_MARCH))
-RISCV_CFLAGS      = $(TEST_RISCV_CFLAGS)
-
-ifeq ($(GNU_YES),YES)
-ifeq ($(call IS_YES,$(TEST_GNU_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the GNU toolchain)
-endif
-RISCV            = $(GNU_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(GNU_VENDOR)-elf-
+# The so-called "COREV Environment Variables".
+#
+# First, check that the COREV environment variables have been defined.
+ifndef CV_SW_TOOLCHAIN
+$(error Must define CV_SW_TOOLCHAIN)
+endif
+ifndef CV_SW_PREFIX
+$(error Must define CV_SW_PREFIX)
+endif
+ifndef CV_SW_MARCH
+$(error Must define CV_SW_MARCH)
+endif
+#ifndef CV_SW_CFLAGS
+#$(warning CV_SW_CFLAGS is optional)
+#endif
+
+# Use these to set defaults for toolchain vars. A few typical examples...
+#RISCV            = /opt/riscv
+#RISCV_PREFIX     = riscv32-unknown-elf-
+#RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
+#RISCV_CFLAGS     =
+RISCV            = $(CV_SW_TOOLCHAIN)
+RISCV_PREFIX     = $(CV_SW_PREFIX)
 RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(GNU_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_GNU_MARCH),$(GNU_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_GNU_CFLAGS),$(GNU_CFLAGS))
-endif
 
-ifeq ($(COREV_YES),YES)
-ifeq ($(call IS_YES,$(TEST_COREV_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the COREV toolchain)
-endif
-RISCV            = $(COREV_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(COREV_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(COREV_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_COREV_MARCH),$(COREV_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_COREV_CFLAGS),$(COREV_CFLAGS))
-endif
+RISCV_CC         = gcc
+RISCV_MARCH      = rv32imc
 
-ifeq ($(PULP_YES),YES)
-ifeq ($(call IS_YES,$(TEST_PULP_NOT_SUPPORTED)),YES)
-$(error test [$(TEST)] does not support the PULP toolchain)
-endif
-RISCV            = $(PULP_SW_TOOLCHAIN)
-RISCV_PREFIX     = riscv32-$(PULP_VENDOR)-elf-
-RISCV_EXE_PREFIX = $(RISCV)/bin/$(RISCV_PREFIX)
-RISCV_CC         = $(PULP_CC)
-RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_PULP_MARCH),$(PULP_MARCH))
-RISCV_CFLAGS     = $(call RESOLVE_FLAG2,$(TEST_PULP_CFLAGS),$(PULP_CFLAGS))
-endif
+RISCV_MARCH      = $(call RESOLVE_FLAG2,$(TEST_RISCV_MARCH),$(COREV_SW_MARCH))
+RISCV_MARCH      = $(call RESOLVE_FLAG2,$(CFG_COREV_MARCH),$(TEST_RISCV_MARCH))
```

Comment:
Does this second assignment always override the first one?

I am not quite sure how to write the intended behavior.
Maybe this?:
```
RISCV_MARCH = rv32imc
RISCV_MARCH = $(call RESOLVE_FLAG2,$(CFG_COREV_MARCH),$(RISCV_MARCH))
RISCV_MARCH = $(call RESOLVE_FLAG2,$(TEST_RISCV_MARCH),$(RISCV_MARCH))
RISCV_MARCH = $(call RESOLVE_FLAG2,$(CV_SW_MARCH),$(RISCV_MARCH))
```

Reply:
@silabs-robin: no, this is not an override per se, it sets the second if the first is not set.

Reply:
Sorry I didn't specify, I meant that the second _line_ always overrides whatever was done in the line before it.
Is it possible for anything other than CFG_COREV_MARCH and TEST_RISCV_MARCH to be used?
The code looks like it can never use the default value nor COREV_SW_MARCH?
And if neither CFG_COREV_MARCH nor TEST_RISCV_MARCH is set, then RISCV_MARCH itself will be empty or undefined?

Reply:
That is what I requested with my comment below, to allow for a prioritized list of variables, s.t. the highest prioritized variable will be selected. 

There is also the option of using ?= instead of = to conditionally set it if the env-var is not set.

Reply:
Agree that this is confusing. This can be simplified if the defaults are _required_ to be set in the cfg-yaml files.